lib/pages/timetable/view/events/add_event_view.dart

```diff
@@ -273,6 +287,60 @@ class _AddEventViewState extends State<AddEventView> {
                               return null;
                             },
                           ),
+                        if ([UniEventType.lecture].contains(selectedEventType))
+                          Autocomplete<Person>(
+                            fieldViewBuilder: (BuildContext context,
+                                TextEditingController textEditingController,
+                                FocusNode focusNode,
+                                VoidCallback onFieldSubmitted) {
+                              textEditingController.text =
+                                  selectedTeacher?.name;
+                              return TextFormField(
+                                controller: textEditingController,
+                                decoration: InputDecoration(
+                                  labelText: S.of(context).labelLecturer,
+                                  prefixIcon: const Icon(Icons.person),
+                                ),
+                                focusNode: focusNode,
+                                validator: (selection) {
+                                  if (selection == null) {
+                                    return S
+                                        .of(context)
+                                        .errorLecturerCannotBeEmpty;
+                                  }
+                                  return null;
+                                },
+                                onFieldSubmitted: (String value) {
+                                  onFieldSubmitted();
+                                },
+                              );
+                            },
+                            displayStringForOption: _displayStringForOption,
```

Comment:
Nit: I'd just inline this instead of defining a named function. (`Person person` => person.name`